docs/sources/next/javascript-api/k6-experimental/redis/client/_index.md

```diff
@@ -1,20 +1,198 @@
 
-`Client` is a Redis client to interact with a Redis server or cluster. It exposes a promise-based API, which users can interact with in an asynchronous manner.
+`Client` is a [Redis](https://redis.io) client to interact with a Redis server, sentinel or cluster. It exposes a promise-based API, which users can interact with in an asynchronous manner.
 
-Though the API intends to be thorough and extensive, it does not expose the whole Redis API.
-Instead, the intent is to expose Redis for use cases most appropriate to k6.
+Though the API intends to be thorough and extensive, it does not expose the whole Redis API. Instead, the intent is to expose Redis for use cases most appropriate to k6.
 
-Note that the `Client` is configured through the [`Options`](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/redis/redis-options) object.
+## Usage
 
-## Example
+### Single-node server
+
+As shown in the above example, the simplest way to create a new `Client` instance that connects to a single Redis server is by passing a URL string.
+It must be in the format:
+
+`redis[s]://[[username][:password]@][host][:port][/db-number]`
+
+Here's an example of a URL string that connects to a Redis server running on localhost, on the default port (6379), and using the default database (0):
+
+{{< code >}}
+
+```javascript
+import redis from 'k6/experimental/redis';
+
+const client = new redis.Client('redis://localhost:6379');
+```
+
+{{< /code >}}
+
+A client can also be instantiated using an [options](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/redis/redis-options) object to support more complex use cases, and for more flexibility:
+
+{{< code >}}
+
+```javascript
+import redis from 'k6/experimental/redis';
+
+const client = new redis.Client({
+  socket: {
+    host: 'localhost',
+    port: 6379,
+  },
+  username: 'someusername',
+  password: 'somepassword',
+});
+```
+
+{{< /code >}}
+
+### TLS
+
+A TLS connection can be established in a couple of ways.
+
+If the server has a certificate signed by a public Certificate Authority, you can use the `rediss` URL scheme:
+
+{{< code >}}
+
+```javascript
+import redis from 'k6/experimental/redis';
+
+const client = new redis.Client('rediss://example.com');
+```
+
+{{< /code >}}
+
+Otherwise, you can supply your own self-signed certificate in PEM format using the [socket.tls](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/redis/redis-options#tls-configuration-options-tlsoptions) object:
+
+{{< code >}}
+
+```javascript
+import redis from 'k6/experimental/redis';
+
+const client = new redis.Client({
+  socket: {
+    host: 'localhost',
+    port: 6379,
+    tls: {
+      ca: [open('ca.crt')],
+    },
+  },
+});
+```
+
+{{< /code >}}
+
+Note that for self-signed certificates, k6's [**insecureSkipTLSVerify**](https://grafana.com/docs/k6/<K6_VERSION>/using-k6/k6-options/reference/#insecure-skip-tls-verify) option must be enabled (set to `true`).
```

Comment:
```suggestion
Note that for self-signed certificates, k6's [insecureSkipTLSVerify](https://grafana.com/docs/k6/<K6_VERSION>/using-k6/k6-options/reference/#insecure-skip-tls-verify) option must be enabled (set to `true`).
```

We generally use bold to highlight UI elements, and italic for emphasis, so you could replace it with italic if you'd like, but I think we might not need it here?

You can also use a [`note` admonition](https://grafana.com/docs/writers-toolkit/write/shortcodes/#admonition) if you'd like to give this part more emphasis.

Reply:
It might also be worth considering `backticks` since it is a code option?

Reply:
👍🏻 

I'll give a try to the note admonition next time then 🙇🏻 

I find the design Grafana applies to backticks in the docs to be lacking contrast, and not really nice to look at, so I prefer to avoid them for this specific cases.

Reply:
> I find the design Grafana applies to backticks in the docs to be lacking contrast, and not really nice to look at, so I prefer to avoid them for this specific cases.

I think that presentation should be secondary to semantics. We can update the website styling to better improve the contrast.

That being said, I'm not certain it's established that all config or code options should be expressed in preformatted text so I'll take that to the Docs style roundtable tomorrow to find out.

Reply:
> I think that presentation should be secondary to semantics.

I disagree, but I do see your point, and I'm fine with it.

> We can update the website styling to better improve the contrast.

Actually just checked back the official documentation and the contrast looks okay. It might have changed in the meantime, or maybe my screen was wrongly calibrated last time I checked (recalibrated recently). I still find that the rounding of the gray background is far too rounded (in my view), and the text slightly too dark though 🙇🏻